skore/src/skore/sklearn/_comparison/precision_recall_curve_display.py

```diff
@@ -0,0 +1,310 @@
+
+                    self.lines_ += self.ax_.plot(
+                        recall,
+                        precision,
+                        color=report_color,
+                        linestyle=class_linestyle,
+                        alpha=0.6,
+                        label=(
+                            f"{report_name} #{report_idx + 1} - class {class_} "
+                            f"(AP = {np.mean(average_precision_class):0.2f})"
+                        ),
+                    )
+
+        xlabel = "Recall"
+        ylabel = "Precision"
+        if info_pos_label:
+            xlabel += info_pos_label
+            ylabel += info_pos_label
+
+        self.ax_.set(
+            xlabel=xlabel,
+            xlim=(-0.01, 1.01),
+            ylabel=ylabel,
+            ylim=(-0.01, 1.01),
+            aspect="equal",
+        )
+
+        if despine:
+            _despine_matplotlib_axis(self.ax_)
+
+        self.ax_.legend(
+            loc="lower right",
+            title=f"{self.ml_task.title()} on $\\bf{{{self.data_source}}}$ set",
+        )
+
+    @classmethod
+    def _from_predictions(
+        cls,
+        y_true: list[list],
+        y_pred: list[list],
+        *,
+        estimators: list,
+        estimator_names: list[str],
+        ml_task,
+        data_source=None,
+        pos_label=None,
+        drop_intermediate=True,
+    ):
+        """Private factory to create a PrecisionRecallCurveDisplay from predictions.
+
+        Parameters
+        ----------
+        y_true : array-like of shape (n_samples,)
+            True binary labels.
+
+        y_pred : array-like of shape (n_samples,)
+            Target scores, can either be probability estimates of the positive class,
+            confidence values, or non-thresholded measure of decisions (as returned by
+            “decision_function” on some classifiers).
+
+        estimators : list of estimator instances
+            The estimators from which `y_pred` is obtained.
+
+        estimator_names : list[str]
+            Name of the estimators used to plot the ROC curve.
+
+        ml_task : {"binary-classification", "multiclass-classification"}
+            The machine learning task.
+
+        data_source : {"train", "test", "X_y"}, default=None
+            The data source used to compute the ROC curve.
+
+        pos_label : int, float, bool or str, default=None
+            The class considered as the positive class when computing the
+            precision and recall metrics.
+
+        drop_intermediate : bool, default=False
+            Whether to drop some suboptimal thresholds which would not appear
+            on a plotted precision-recall curve. This is useful in order to
+            create lighter precision-recall curves.
+
+        Returns
+        -------
+        display : PrecisionRecallCurveDisplay
+        """
+        estimator_classes = map(attrgetter("classes_"), estimators)
```

Comment:
```suggestion
        estimator_classes = [est.classes_ for est in estimators]
```

It is a legit code but you will have the following issue:

- The Zen of Python will tell you that the readability count and the map/reduce is usually replaced by a list comprehension if memory is not an issue
- Related to the previous one, `map` will return a generator. Thus, we will not raise an error before actually consuming the iterator and it could be inside a part of the code where it will be easy to understand what `classes_` is.
- Because we have a generator, if we consume it once, then we cannot access the classes a second time. So it could be prone to error in this specific case because we don't really care about the memory footprint but we might need to access several time.

Reply:
> The Zen of Python will tell you that the readability count and the map/reduce is usually replaced by a list comprehension if memory is not an issue

This is the equivalent of emacs/vim debate, there is pros and cons, i use one or the other depending on the situation.
Otherwise, the code assumes that we only have to consume one time the classes of estimator.
This is why i use a generator.

No worries to change, fixed for you in 2ed61d0b4a260ce157551f79e8905f4161175670.

Reply:
Just keep in mind that generator is ideal when you want to make your code more readable without overhead.